### PR TITLE
Proofread docs and expand bespoke-workflows

### LINKED ISSUE #139: Allow gateway port to be configured in CLI
The default gateway port is 8000, which is a commonly used port for HTTP servers. I just got this error trying to run BespokeFit:
```
Traceback (most recent call last):
  File "/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/bin/openff-bespoke", line 33, in 
<module>
    sys.exit(load_entry_point('openff-bespokefit', 'console_scripts', 'openff-bespoke')())
  File 
"/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/lib/python3.9/site-packages/click/core.py", 
line 829, in __call__
    return self.main(*args, **kwargs)
  File 
"/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/lib/python3.9/site-packages/click/core.py", 
line 782, in main
    rv = self.invoke(ctx)
  File 
"/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/lib/python3.9/site-packages/click/core.py", 
line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File 
"/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/lib/python3.9/site-packages/click/core.py", 
line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File 
"/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/lib/python3.9/site-packages/click/core.py", 
line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File 
"/home/joshmitchell/conda/envs/bespokefitdevdocsupdate/lib/python3.9/site-packages/click/core.py", 
line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/joshmitchell/Documents/openff/bespokefit/openff/bespokefit/cli/executor/run.py", line 
53, in _run_cli
    with BespokeExecutor(
  File "/home/joshmitchell/Documents/openff/bespokefit/openff/bespokefit/executor/executor.py", line
399, in __enter__
    self._start(asynchronous=True)
  File "/home/joshmitchell/Documents/openff/bespokefit/openff/bespokefit/executor/executor.py", line
346, in _start
    wait_for_gateway()
  File 
"/home/joshmitchell/Documents/openff/bespokefit/openff/bespokefit/executor/services/gateway.py", 
line 114, in wait_for_gateway
    raise RuntimeError("The gateway could not be reached.")
RuntimeError: The gateway could not be reached.
```

Turns out this was because the gateway port was already in use. The other software I was using allows the port to be configured so this wasn't a huge issue, but we should add a `--gateway-port` and possible a `--redis-port` CLI flag to allow this to be changed.

### PR DESCRIPTION
**Pull request by Yoshanuikabundi**
## Description
This is just a quick once-over the Book-style docs for good englishing, consistent spelling, and clarity. I've also expanded the discussion on workflows to be more complete and more consistent with the new theory docs

Update: I've added a section to the executor docs that will automatically document that environment variables for the executor. I haven't added docstrings, but they can be added to the Settings object and they will appear there.

Closes #139 

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by Yoshanuikabundi** [comment]
I've added a section to the executor docs that will automatically document that environment variables for the executor. I haven't added docstrings, but they can be added to the Settings object and they will appear there.

Closes #139

File: docs/users/bespoke-workflows.md [hunk 1]
```diff
@@ -12,104 +12,149 @@ bespoke fit (represented by the [`BespokeOptimizationSchema`] object), and are u
...
+field) with a new bespoke torsion term for each non-terminal rotatable bond in the input molecule, trained to reproduce
+a bespoke one-dimensional quantum chemical torsion scan performed around that bond. 
 
 [SMIRNOFF]: https://openforcefield.github.io/standards/standards/smirnoff/
 
 In this section, we'll build this workflow factory up from nothing. This is just for a demonstration; you can
 always build this factory by simply instantiating the `BespokeWorkflowFactory` class without arguments.
 
-The default workflow has three steps:
+The default workflow has five steps:
```
Comment (j-wags) [inline docs/users/bespoke-workflows.md:24]: This change makes sense - It lines up with the 5 steps in the theory guide

File: docs/users/bespoke-workflows.md [hunk 2]
```diff
@@ -12,104 +12,149 @@ bespoke fit (represented by the [`BespokeOptimizationSchema`] object), and are u
...
 
-1. **Fragmentation:** Break the molecule into parts large enough to have accurate
-chemistry, but small enough for efficient quantum chemical calculations.
-2. **QC Generation:** Perform quantum chemical calculations on the fragments to
-generate reference data for the bespoke parameters to reproduce.
-3. **Optimization:** Optimize the force field parameters into bespoke parameters
-that reproduce the quantum chemical data.
+1. **Parameter selection:** Choose the features of the molecule to create bespoke 
+    parameters for
```
Comment (j-wags) [inline docs/users/bespoke-workflows.md:27]:
```suggestion
1. **Parameter selection:** Choose the features of the molecule to create bespoke 
    parameters for.
```

**Review by j-wags — APPROVED** [review]
Looks great. I pushed a commit a suggested a change or two but for the most part this looks totally awesome. Thanks for the detailed look and careful work @Yoshanuikabundi!

**Comment by mattwthompson** [comment]
Would you consider adding a reference in the README to improve discoverability of the examples? Right now it's not clear without clicking through to the docs that there are such detailed and thorough examples. I'm thinking just a one-liner like

```
### Examples

Usage examples covering X Y Z can be found in the [user guide](https://docs.openforcefield.org/projects/bespokefit/en/latest/index.html) section of the docs.
```

Yesterday I did a search for "example" at the repo's homepage and, getting no hits, I thought that the examples were not yet complete and didn't look any further (I'm lazy, yes, but some users are as well). I won't argue against the idea that users should do some digging for themselves, but I think a little bit of discoverability can go a long way.

**Comment by j-wags** [comment]
Totally agree with the need to use the word "example" somewhere - I've done that search too and come to the same conclusion. I've pushed an update to the README to have an instance of the word "example", and had the sentence link straight to the quick start guide.

**Pull request merged**